partner_contact_age_range/models/res_partner_age_range.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartnerDateRange(models.Model):
+    _name = "res.partner.age.range"
+    _description = "Partner Age Range"
+
+    name = fields.Char(string="Name", required=True)
+    age_from = fields.Integer(string="From", required=True)
+    age_to = fields.Integer(string="To", required=True)
+
+    _sql_constraints = [("name_uniq", "unique (name)", "A name must be unique !")]
+
+    @api.constrains("age_from", "age_to")
+    def _validate_range(self):
+        for rec in self:
+            if rec.age_from > rec.age_to:
+                raise ValidationError(
+                    _("%s is not a valid range (%s > %s)")
+                    % (rec.name, rec.age_from, rec.age_to)
+                )
+            range_id = rec.search(
+                [
+                    ("age_from", "<=", rec.age_to),
+                    ("age_to", ">", rec.age_from),
+                    ("id", "!=", rec.id),
+                ],
+                limit=1,
+            )
```

Comment:
This constraint shouldn't allow this either:
![image](https://user-images.githubusercontent.com/1914185/81150457-35d5a080-8f80-11ea-9602-2f31cc8aadf9.png)

A res.partner with age=45 would raise an error:

```
ValueError: Expected singleton: res.partner.age.range(4, 6, 7)
```

Reply:
@ivantodorovich : I have fix constraint